operator/v1/types_etcd.go

```diff
@@ -38,6 +38,11 @@ type EtcdSpec struct {
 	// +openshift:enable:FeatureSets=CustomNoUpgrade;TechPreviewNoUpgrade
 	// +optional
 	HardwareSpeed ControlPlaneHardwareSpeed `json:"controlPlaneHardwareSpeed"`
+
+	// EtcdDBSize allows user to customize the etcd database backend storage size.
+	// +kubebuilder:validation:Optional
+	// +optional
+	EtcdDBSize *int64 `json:"etcdDBSize"`
```

Comment:
Let's hold off on the exact naming and placement of this field until we have a chance to put up the enhancement for review first.
E.g this will likely be a tech-preview field like the tuning parameters.